### ISSUE TITLE
[clang][modules] False positive type mismatch error when a "typedef enum" is used through C++20 modules

### ISSUE THREAD

**Issue by usachev63** [llvm/llvm-project#213299]
A minimized example: [clang-bug.tar.gz](https://github.com/user-attachments/files/30595495/clang-bug.tar.gz)

`my_enum.h`
```cpp
#pragma once

typedef enum {
  MY_ENUM_0,
} my_enum;
```

`my_module.cppm`

```cpp
module;

#include "my_enum.h"

export module my_module;

export my_enum glob;
```

`bar.cpp`

```cpp
import my_module;

#include "my_enum.h"

inline void bar() { foo(MY_ENUM_0); }
```

```sh
clang++-23 -std=c++23 -fmodules -fmodule-output=my_module.pcm -c my_module.cppm
clang++-23 -std=c++23 -fmodules -fmodule-file=my_module=my_module.pcm -c bar.cpp
```

This produces a funny error:
```
bar.cpp:5:21: error: no matching function for call to 'foo'
    5 | inline void bar() { foo(MY_ENUM_0); }
      |                     ^~~
./my_enum.h:7:6: note: candidate function not viable: no known conversion from 'my_enum' to 'my_enum' (aka 'enum my_enum') for 1st argument
    7 | void foo(my_enum e);
      |      ^   ~~~~~~~~~
1 error generated.
```

- Reproduced with Clang 21, 22, 23, but not with 20. I got Clang from the apt repository. Ubuntu 25.10, also reproduced with Clang 22 on Debian 13.
- Does not reproduce if we say `enum my_enum {...}` instead of `typedef enum {...} my_enum`;
- Does not reproduce without `my_enum` mention in `my_module.cppm`.
